pkg/testing/runner/ssh.go

```diff
@@ -189,12 +222,21 @@ func (s *sshClient) Exec(ctx context.Context, cmd string, args []string, stdin i
 		return nil, nil, ctx.Err()
 	}
 
+	var session *ssh.Session
 	cmdArgs := []string{cmd}
 	cmdArgs = append(cmdArgs, args...)
 	cmdStr := strings.Join(cmdArgs, " ")
 	session, err := s.NewSession()
 	if err != nil {
-		return nil, nil, fmt.Errorf("could not create new SSH session: %w", err)
+		s.logger.Logf("new session failed: %q, trying reconnect", err)
+		lErr := s.Reconnect(ctx)
+		if lErr != nil {
+			return nil, nil, fmt.Errorf("ssh reconnect failed: %w, after new session failed: %w", lErr, err)
+		}
+		session, lErr = s.NewSession()
+		if lErr != nil {
+			return nil, nil, fmt.Errorf("new session failed after reconnect: %w, original new session failure was: %w", lErr, err)
+		}
```

Comment:
If understand this correctly, we have only one retry without a delay here, should we use a configurable backoff/retry here?

Reply:
The retry isn't obvious, `Reconnect` calls `Close` then `Connect`, and the `Connect` has the retry with exponential back off.  If `NewSession` fails, that means we failed twice after what should have been valid connections via `Connect`, at that point I think it is good to just print the errors and let a human figure out what is going on.